### ISSUE TITLE
[question]reward for Humanoid-v2 (using ddpg)does not increase after 300

### ISSUE THREAD

**Issue by qwang029-wqy** [hill-a/stable-baselines#858]
hi   

My reward stays at 300 after 100w steps whereas yours increases almost linearly.  My actor loss is between -4.5 and -0.5 while my critic loss is between 0 and 0.06, which is way much smaller than yours. The man cannot walk at all.

We have similar settings with yours (mlpPolicy.env with action noise)  in hyper-parameter, model structure, activation function, buffer size, 100 steps of interaction after training  50 steps.
The only difference is the optimizer (mine is adam). 

Would you be able to provide some suggestions? Are there any other things I can change to make it work? Is the small loss and the optimizer the key issue?  
ps: my codes work perfectly fine for Pendulum-v0.

Thanks~~

![WechatIMG3199](https://user-images.githubusercontent.com/60876718/81942641-f4528e80-962c-11ea-8eee-1768f978984a.png)

[my code]
part1: interacte 100 episodes and train 50 steps:
`
scores_deque = deque(maxlen=100)
list_j = deque(maxlen=100)

def trainTest():
    env = gym.make("Humanoid-v2")
    env = env.unwrapped
    env.seed(1)

    n_actions = env.action_space.shape[-1]
    # VAR = 0.5
    action_noise = OrnsteinUhlenbeckActionNoise(mean=np.zeros(n_actions), sigma=float(0.5) * np.ones(n_actions))

    k = 0
    action_noise.reset()
    s = env.reset()
    ep_reward = 0
    j_p = 0

    for i in range(200000):

        for j in range(100):
            k += 1
            noise = 0.4*action_noise()
            print("noise",noise)

            a = model.choose_action(s)
            a = np.clip(a+noise, -0.4, 0.4)  # add randomness to action selection for exploration
            s_, r, done, info = env.step(a)
            M.store_transition(s, a, r , s_,float(done))
            s=s_
            ep_reward+=r
            j_p+=1
            if done:
                scores_deque.append(ep_reward)
                # print(j)
                # if scores_deque:
                #     m = max(scores_deque)
                #     if (ep_reward >= m):
                #         model.model_save('gym_hum_walk_test')
                list_j.append(j_p)
                j_p=0
                ep_reward=0
                action_noise.reset()
                s = env.reset()

        for m in range(50):

            if M.pointer > Memory_SIZE:
                data_batch = M.sample(mp.ddpgParams[vn.BATCH_SIZE])
                # print("m",m)
                model.learn(data_batch, (k-100)+2*m-50000,scores_deque,noise)
            else:
                break
`
part2: actor and critic net 
`
def build_actor_net(self, s, scope, trainable,normal_flag=False):
        with tf.variable_scope(scope):
            init_w = tf.random_uniform_initializer(minval=-3e-3,maxval=3e-3)
            init_b = tf.constant_initializer(0.1)
            n_l1=64
            w1_s = tf.get_variable('a_w1_s', [self.modelParams[vn.STATE_DIM], n_l1], initializer=init_w,
                                   trainable=trainable)
            b1 = tf.get_variable('a_b1', [1, n_l1], initializer=init_b, trainable=trainable)
            # net = tf.layers.dense(s, 400, activation=tf.nn.tanh,
            #                       kernel_initializer=init_w, bias_initializer=init_b, name='l1',
            #                       trainable=trainable)
            l1_no_active=tf.matmul(s,w1_s)+b1

            if normal_flag:
                l1_net_normal = self.normalize_layer(l1_no_active, n_l1)
            else:
                l1_net_normal=l1_no_active
            l1_output=self.activate_f(l1_net_normal)
            with tf.variable_scope('a'):
                n_l2=64
                w2_s = tf.get_variable('a_w2_s', [n_l1, n_l2], initializer=init_w,
                                       trainable=trainable)
                b2 = tf.get_variable('a_b2', [1, n_l2], initializer=init_b, trainable=trainable)

                l2_no_active = tf.matmul(l1_output, w2_s) + b2
                # print (l1_output,w2_s,b2,l2_no_active)
                if normal_flag:
                    l2_net_normal = self.normalize_layer(l2_no_active, n_l2)
                else:
                    l2_net_normal = l2_no_active

                l2_output=self.activate_f(l2_net_normal)

                n_l3 = self.modelParams[vn.ACTION_DIM]
                w3_s = tf.get_variable('a_w3_s', [n_l2, n_l3], initializer=init_w,
                                       trainable=trainable)
                b3 = tf.get_variable('a_b3', [1, n_l3], initializer=init_b, trainable=trainable)

                l3_no_active = tf.matmul(l2_output, w3_s) + b3

                if normal_flag:
                    l3_net_normal = self.normalize_layer(l3_no_active, n_l3)
                else:
                    l3_net_normal = l3_no_active

                action_output = tf.nn.tanh(l3_net_normal)

                tf.summary.histogram('a_l1_output', l1_output)
                # tf.summary.histogram('l2_no_active', l2_no_active)
                # tf.summary.histogram('l2_net_normal', l2_net_normal)
                # print( self.modelParams[vn.ACTION_BOUND])
                scaled_a = tf.multiply(action_output, self.modelParams[vn.ACTION_BOUND],
                                   name='scaled_a')  # Scale output to -action_bound to action_bound

                return scaled_a

    def build_critic_net_ly2(self, s, a, scope, trainable,normal_flag=False):
        with tf.variable_scope(scope):
            init_w = tf.random_uniform_initializer(minval=-3e-3,maxval=3e-3)
            init_b = tf.constant_initializer(0.1)

            with tf.variable_scope('critic_l1'):
                n_l1 = 64
                w1_s = tf.get_variable('c_w1_s', [self.modelParams[vn.STATE_DIM], n_l1], initializer=init_w,
                                       trainable=trainable)

                b1 = tf.get_variable('c_b1', [1, n_l1], initializer=init_b, trainable=trainable)

                l1_out_noact=tf.matmul(s, w1_s) + b1

                if normal_flag:
                    l1_normal = self.normalize_layer(l1_out_noact, n_l1)
                else:
                    l1_normal = l1_out_noact

                l1_output = self.activate_f(l1_normal)

            with tf.variable_scope('critic_l2'):
                n_l2 = 64
                # l2_output = tf.layers.dense(l1_output, n_l2, kernel_initializer=init_w, bias_initializer=init_b,
                #                             trainable=trainable)

                l1_a = tf.concat([l1_output, a], axis=-1)
                w2_s = tf.get_variable('c_w2_s', [n_l1+self.modelParams[vn.ACTION_DIM], n_l2], initializer=init_w,
                                       trainable=trainable)

                
                #                        trainable=trainable)
                b2 = tf.get_variable('c_b2', [1, n_l2], initializer=init_b, trainable=trainable)
                l2_out_noact = tf.matmul(l1_a, w2_s)  + b2

                if normal_flag:
                    l2_net_normal = self.normalize_layer(l2_out_noact, n_l2)
                else:
                    l2_net_normal = l2_out_noact

                l2_output = self.activate_f(l2_net_normal)

            with tf.variable_scope('q'):
                q = tf.layers.dense(l2_output, 1, kernel_initializer=init_w, bias_initializer=init_b,
                                    trainable=trainable)  # Q(s,a)

            tf.summary.histogram('c_l1_output', l1_output)
            tf.summary.histogram('c_l2_output', l2_output)
            tf.summary.histogram('q_output', q)
        return q
`

part3:noise
`action_noise = OrnsteinUhlenbeckActionNoise(mean=np.zeros(n_actions), sigma=float(0.5) * np.ones(n_actions))
`

[the settings for yours]
`env = gym.make("Humanoid-v2")
env = env.unwrapped
env.seed(1)

# the noise objects for DDPG
n_actions = env.action_space.shape[-1]
param_noise = AdaptiveParamNoiseSpec()
action_noise = OrnsteinUhlenbeckActionNoise(mean=np.zeros(n_actions), sigma=float(0.5) * np.ones(n_actions))

action_noise.reset()

model = DDPG(MlpPolicy, env, verbose=1, param_noise=None, action_noise=action_noise,tensorboard_log='tb_data_baseline',full_tensorboard_log=True)

model.learn(300000)
`

**Comment by Miffyli** [comment]
Did you try the parameters from [rl-zoo](https://github.com/araffin/rl-baselines-zoo)? I am also slightly confused: When you say "my code works on Pendulum", do you mean the same code using stable-baselines works on Pendulum but does not work on Humanoid?

**Comment by araffin** [comment]
>We have similar settings with yours

Please fill the issue template completely. It sounds like your are using a custom DDPG implementation?

**Comment by qwang029-wqy** [comment]
@araffin    appologies for the format . It's my first time asking question and adding code ....l will upload the complete version and send out the link tomorrow (it's now nearly midnight here in China.)

**Comment by qwang029-wqy** [comment]
> Did you try the parameters from [rl-zoo](https://github.com/araffin/rl-baselines-zoo)? There seems to be no parameters for Humanoid env specifically, but you could take the closest match to see how that works.
> by  saying "my code works on Pendulum", i mean my code, not the stable baselines.
> I am also slightly confused: When you say "my code works on Pendulum", do you mean the same code using stable-baselines works on Pendulum but does not work on Humanoid?
> 
> Edit: Possibly related discussion/points #837

by

**Comment by qwang029-wqy** [comment]
@hill-a  sorry for the delay. I got some progress! my reward has become nearly 800. The changes are:

-  increasing gamma in calculating q value from 0.9 to 0.99 

- adding larger noises on actons.

my questions are: 

1. in which situation should we add more emphasis on instant reward (larger gamma)? 0.9 works fine in Pendulum.
2. why are noises so important ? how much should it be compared to the original action 
3. it looks like we should never expect critic loss to converge very quickly, should we ?
<img width="730" alt="WechatIMG3386" src="https://user-images.githubusercontent.com/60876718/82214910-1871e080-9949-11ea-8a29-d67b3ebb25bb.png">

**Comment by araffin** [comment]
Closing this as it is apparently unrelated to SB.
